src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewList.cs

```diff
@@ -0,0 +1,687 @@
+			{
+				var tvItem = (TreeViewItem)ContainerFromItem(dragItem);
+				m_draggedTreeViewNode = NodeFromContainer(tvItem);
+				bool isMultipleDragging = false;
+				if (IsMutiSelectWithSelectedItems)
+				{
+					int selectedCount = ListViewModel.SelectedNodes.Count;
+					if (selectedCount > 1)
+					{
+						var settings = tvItem.TreeViewItemTemplateSettings;
+						settings.DragItemsCount = selectedCount;
+						isMultipleDragging = true;
+
+						// TreeView items' selection states are maintained by ViewModel, not by TreeViewList.
+						// When TreeView is set to multi-select mode, the underlying TreeViewList's selection mode is set to None (See OnPropertyChanged function in TreeView.cpp).
+						// TreeViewList has no knowledge about item selections happened in TreeView, no matter how many items are actually selected, args.Items() always contains only one item that is currently being dragged by cursor.
+						// Here, we manually add selected nodes to args.Items in order to expose all selected (and dragged) items to outside handlers.
+						args.Items.Clear();
+
+						foreach (var node in ListViewModel.SelectedNodes)
+						{
+							if (IsContentMode)
+							{
+								args.Items.Append(node.Content);
+							}
+							else
+							{
+								args.Items.Append(node);
+							}
+						}
+					}
+				}
+				else
+				{
+					// If not in multi-select mode, collapse the dragged node.
+					m_draggedTreeViewNode.IsExpanded = false;
+				}
+
+				VisualStateManager.GoToState(
+					tvItem,
+					isMultipleDragging ? "MultipleDraggingPrimary" : "Dragging",
+					false);
+				UpdateDropTargetDropEffect(false, false, null);
+			}
+		}
+
+		private void OnDragItemsCompleted(object sender, DragItemsCompletedEventArgs args)
+		{
+			m_draggedTreeViewNode = null;
+			m_emptySlotIndex = -1;
+		}
+
+		private void OnContainerContentChanging(object sender, ContainerContentChangingEventArgs args)
+		{
+			//	// Only need to update unrecycled on screen items.
+			if (!args.InRecycleQueue)
+			{
+				var targetItem = (TreeViewItem)args.ItemContainer;
+				var targetNode = NodeFromContainer(targetItem);
+				var treeViewItem = targetItem;
+				treeViewItem.UpdateIndentation(targetNode.Depth);
+				treeViewItem.UpdateSelectionVisual(targetNode.SelectionState);
+			}
+		}
+
+		// IControlOverrides
+		protected override void OnDrop(DragEventArgs e)
+		{
+			DragEventArgs args = e;
+
+			if (args.AcceptedOperation == DataPackageOperation.Move && !args.Handled)
+			{
+				if (m_draggedTreeViewNode != null && IsIndexValid(m_emptySlotIndex))
+				{
+					// Get the node at which we will insert
+					TreeViewNode insertAtNode = NodeAtFlatIndex(m_emptySlotIndex);
+
+					if (IsMutiSelectWithSelectedItems)
+					{
+						// Multiselect drag and drop. In the selected items, find all the selected subtrees 
+						// and move each of those subtrees.
+						var selectedRootNodes = GetRootsOfSelectedSubtrees();
+						var selectionSize = selectedRootNodes.Count;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'selectionSize'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)